app/api/dds/v1/files_api.rb

```diff
@@ -21,17 +21,23 @@ class FilesAPI < Grape::API
         authorize DataFile.new(project: project), :download?
         files = project.data_files.unscope(:order).order(updated_at: :desc).where(is_deleted: false)
 
-        # plain query
-        #paginate(files)
+        files_query = headers&.fetch("Project-Files-Query", nil) || "plain"
```

Comment:
@dmlond We can use this `Project-Files-Query` header when we profile this endpoint, sort of like a feature flag. We should probably remove this logic once we've determined which query is optimal.

Reply:
good idea